### PR TITLE
feat(generator): remove unused links in generated main.dox

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7859)

### CODE REVIEW

File: generator/internal/scaffold_generator.cc [hunk 1]
```diff
@@ -628,9 +627,7 @@ which should give you a taste of the $title$ C++ client library API.
   the program's console.
 
 - `GOOGLE_CLOUD_CPP_ENABLE_TRACING=rpc-streams` turns on tracing for streaming
-  gRPC calls, such as `Client::Read()`, `Client::ExecuteQuery()`, and
-  `Client::ProfileQuery()`. This can produce a lot of output, so use with
-  caution!
+  gRPC calls. This can produce a lot of output, so use with caution!
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:630]: It's annoying, but I think you need to make this change in the following files:
```sh
# grep -rl "gRPC calls, such as" --exclude "spanner-main.dox" | sort
accessapproval/doc/main.dox
assuredworkloads/doc/main.dox
billing/doc/main.dox
kms/doc/main.dox
logging/doc/main.dox
pubsublite/doc/main.dox
scheduler/doc/main.dox
secretmanager/doc/main.dox
tasks/doc/main.dox
webrisk/doc/main.dox
websecurityscanner/doc/main.dox
workflows/doc/main.dox
```

File: generator/internal/scaffold_generator.cc [hunk 2]
```diff
@@ -668,15 +665,9 @@ can override the default policies.
 
 [cloud-service-docs]: https://cloud.google.com/$site_root$
 [exponential backoff]: https://en.wikipedia.org/wiki/Exponential_backoff
-[gcs-quickstart]: https://cloud.google.com/$site_root$/docs/quickstarts 'Quickstarts'
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:671]: less important, but these changes could be propagated too

File: generator/internal/scaffold_generator.cc (hunk 1 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc:630]: Done.

File: generator/internal/scaffold_generator.cc (hunk 2 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc:671]: Done.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**